### ISSUE TITLE
Add an Action endpoint route that supports Action as State Changes

### ISSUE THREAD

**Issue by zhengtr** [CSCI-GA-2820-SP23-003/inventory#27]
**As a** Inventory Manager
**I need** an Action endpoint route
**So that** I can make an action based call on an Inventory Entry as State Changes
   
 ### Details and Assumptions
 * We will store the inventory details in a database
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given the service is running
 When I make a PUT request "PUT http://.../Inventory/<inventory_id> { ..., <specified_state>: <new_state> }"
 Then the <specified_state> of an Inventory entry with id=<inventory_id> will be changed to <new_state>.
```

**Comment by sahildeshpande25** [comment]
Action - restock
Should update the quantity of an item to `restock_level + 1`
Should return a 409 conflict when called on an item whose quantity is above the restock level

**Comment by zhengtr** [comment]
Note: this issue is closed because the only action we need is `restock`, which is included in issue #37.
